### ISSUE TITLE
Able to Build, Unable to Map Keys

### ISSUE THREAD

**Issue by claytonsulby** [Jamal-Ra-Davis/Guardians_of_Sunshine#3]
My build is able to load, but I am unable to map the keys of the controller I am using to the keys listed in the README.md. I am looking into the src but I cannot locate the exact instances of the keys. I do see the key enum in InputBuffer.h. Is there an exact location?

OS: RetroPie 4.6
HW: Raspberry Pi CM3

**Comment by Jamal-Ra-Davis** [comment]
The keys get mapped inside GameState.cpp in the keyPressed() and keyReleased() functions

**Comment by claytonsulby** [comment]
Thank you! Are you familiar with EmulationStation? It's supposed to have been built on top of SDL. I've been attempting to map the SDL Game Controller buttons to where you used SDLK but I've had no luck.

**Comment by Jamal-Ra-Davis** [comment]
hmm... I don't know if emulation station's setting get passed along to the game once it's running or not. You may need something that runs in the background that reads in the gamepad button presses and issues the equivalent keyboard key presses.

I suppose another option would be to modify the game to take in the gamepad inputs directly, but I'm not really familiar with how that would be implemented.
https://wiki.libsdl.org/CategoryGameController

**Comment by claytonsulby** [comment]
That was super helpful! I was able to open up a joystick and map the hat (dpad) and the buttons to your key pressed and released functions. I had to implement separate button and hat functions but I also changed the input handler function. I could probably figure out how to get both working side by side though!

I used the info on this website too: https://www.libsdl.org/release/SDL-1.2.15/docs/html/guideinput.html
